sklearn/tree/_export.py

```diff
@@ -259,7 +260,12 @@ def get_fill_color(self, tree, node_id):
             self.colors["rgb"] = _color_brew(tree.n_classes[0])
             if tree.n_outputs != 1:
                 # Find max and min impurities for multi-output
-                self.colors["bounds"] = (np.min(-tree.impurity), np.max(-tree.impurity))
+                # Avoid what looks like an issue with SIMD on non memory
+                # aligned arrays on 32bit OS, see
+                # https://github.com/scikit-learn/scikit-learn/issues/27506 for
+                # more details
+                minus_impurity = -1.0 * tree.impurity if _IS_32BIT else -tree.impurity
+                self.colors["bounds"] = (np.min(minus_impurity), np.max(minus_impurity))
```

Comment:
would the following work ? It avoids 1 unnecessary array multiplication and it avoids to specialize the behavior for 32bit OS.
```suggestion
                self.colors["bounds"] = (-np.max(tree.impurity), -np.min(tree.impurity))
```

Reply:
That was my original thought in https://github.com/scikit-learn/scikit-learn/issues/27506#issuecomment-2157554782 :wink:.

I find it slightly too clever and I kind of doubt this matters that much in practice since this is not in a performance-critical part of the code (visualization/debugging)

Reply:
It's a matter of taste indeed. I find it more intuitive.
One argument: the workaround should be removed when the bug is fixed upstream, while the suggestion can be let as is for ever

Reply:
@jeremiedbb OK you won me over by talking to my ex-physicist side :wink:, I have added a comment to kind of indicate there is a reason why we write it like this.

I have also added an entry in the 1.5.1 changelog.